tools/validators/instance_validator/parser.py

```diff
@@ -0,0 +1,56 @@
+from strictyaml import Map, MapPattern, Str, Optional, Any, load, Enum, Regex, Seq, YAMLValidationError
+
+# TODO check all valid states and ontological references in next validation steps
+schema = MapPattern(Str(), 
+    Map({
+        'type': Str(), 
+        'id': Str(), 
+        Optional('connections'): MapPattern(Str(), Str())
+                               | Seq(MapPattern(Str(), Str())),
+        Optional('links'): MapPattern(Str(), 
+            MapPattern(Str(), Str())),
+        Optional('translation'): Any(),
+        Optional('metadata'): Any()
+    }))
+
+translation_schema = Str() | Any()
+
+# TODO add manual check for translation_data_schema to de-duplicate units/unit_values/states
+# TODO add all units/unit_values/states to translation_data_schema
+translation_data_schema = Str() | Map({
+                                    'present_value': Str(),
+                                    Optional('states'): MapPattern(Str(), Str()),
+                                    Optional('units'): Map({
+                                        'key': Str(),
+                                        'values': MapPattern(Str(), Str())
+                                    }),
+                                    Optional('unit_values'): MapPattern(Str(), Str())
+                                })
+
+def load_yaml_with_schema(filepath, schema):
+    f = open(filepath).read()
+
+    try:
+        parsed = load(f, schema)
+        return parsed
+    except YAMLValidationError as error:
+        raise error
+
+def parse_yaml(filename: str):
+    yaml = load_yaml_with_schema(filename, schema)
+
+    top_name = yaml.keys()[0]
+
+    if 'translation' in yaml[top_name].keys():
+        translation = yaml[top_name]['translation']
```

Comment:
I don't follow what's happening here.  How would a file ever have "translation' as a top level key?

Reply:
`yaml[top_name]` on `line 63` accesses the first entity in the YAML file, and "translation" may be a top level key in the entity.

My code is assuming that there's only one main building entity in the YAML file so it only accesses the first entity with `yaml.keys()[0]`. If this assumption is wrong, I'll change the code to validate all the parsed entities?